core/src/forest/ForestTrainer.cpp

```diff
@@ -136,13 +135,15 @@ std::vector<std::shared_ptr<Tree>> ForestTrainer::train_ci_group(Data* data,
   std::vector<std::shared_ptr<Tree>> trees;
 
   std::vector<size_t> sample;
-  sampler.sample(data->get_num_rows(), 0.5, sample);
+  std::vector<size_t> dummy_oob_sample;
```

Comment:
While this PR was open, I refactored `RandomSampler` to not accept 'OOB samples' unless it is really necessary. As you update your sampling approach, it be good to remove unnecessary `oob_sample` parameters as well.